ketcher-autotests/tests/Reaction-tool/Mapping-Unmapping tool/mapping-tool.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { test } from '@playwright/test';
+import {
+  delay,
+  takeEditorScreenshot,
+  LeftPanelButton,
+  clickInTheMiddleOfTheScreen,
+  selectLeftPanelButton,
+  selectNestedTool,
+  ReactionMappingTool,
+  DELAY_IN_SECONDS,
+  openFileAndAddToCanvas,
+  selectRingButton,
+  RingButton,
+  getCoordinatesTopAtomOfBenzeneRing,
+  selectTopPanelButton,
+  TopPanelButton,
+  dragMouseTo,
+} from '@utils';
+import { getAtomByIndex } from '@utils/canvas/atoms';
+
+test.describe('Open Ketcher', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('');
+  });
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  test('Click atoms to map atoms in a reaction', async ({ page }) => {
+    /* Test case: EPMLSOPKET-1799, EPMLSOPKET-8909
+    Description:  Click atoms to map atoms in a reaction
+    */
+    await openFileAndAddToCanvas('reaction_3.rxn', page);
+    await selectNestedTool(page, ReactionMappingTool.MAP);
+    await page.getByText('CEL');
+    const { x, y } = await getCoordinatesTopAtomOfBenzeneRing(page);
+    await dragMouseTo(x, y, page);
+  });
+
+  test.describe('mapped_atoms.rxn', () => {
+    test.beforeEach(async ({ page }) => {
+      await openFileAndAddToCanvas('mapped_atoms.rxn', page);
+    });
+
+    test('Click the single mapped atom to delete mapping', async ({ page }) => {
+      // EPMLSOPKET-1827
+      await selectNestedTool(page, ReactionMappingTool.UNMAP);
+      await delay(DELAY_IN_SECONDS.FIVE);
```

Comment:
Let's discuss later, why do we need delay? Every delay is potential problem, which may occur in any time. Please, use `delay` as a LAST RESORT